### PR TITLE
Efacilityregister validations

### PR DESCRIPTION
**Pull request by lytonnyemba**
validations made

### CODE REVIEW

**Review by nthezemu — APPROVED** [review]
There are some conflicts hence the branch can not be merged automatically. we have to do it manually

File: app/controllers/facilities_controller.rb [hunk 1]
```diff
@@ -29,9 +32,10 @@ def create
         phone_number: params[:facility][:phone_number],
         latitude: params[:facility][:longitude],
         longitude: params[:facility][:latitude],
-        type_code: params[:facility][:type_code],
```
Comment (liwoo) [inline app/controllers/facilities_controller.rb:28]: what happens when you say `params[:typename][0]`, then you wouldn't need to have that temporary `type_name_value` up there...

File: app/controllers/facilities_controller.rb [hunk 2]
```diff
@@ -17,6 +17,9 @@ def get_dependencies
 
 
    def create
+      type_name = []
+      type_name = params[:typename]
+      type_name_value = type_name[0]
       @facility = Facility.new(
```
Comment (liwoo) [inline app/controllers/facilities_controller.rb:23]: @nthezemu you may want to use Facility.create a shortcut to make it read:

```
if Facility.create(params)
  #dosomethings
end
```

File: app/models/facility_resource.rb [hunk 3]
```diff
@@ -7,6 +7,9 @@ def self.get_resources_params(params)
       end	
   end
   def self.update_resources_details(facility_id,resources)
+      if(resources.blank?)
```
Comment (liwoo) [inline app/models/facility_resource.rb:10]: good practice is to say `if resources.blank?`

**Review by liwoo — COMMENTED** [review]
by breaking down into smaller models, the code is more manageable and readable.  Also by pushing most logic into the models, the controller is looking cleaner, good job @lytonnyemba

**Pull request merged**